src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -74,19 +82,40 @@ public function count()
      */
     public function add($name, Route $route)
     {
-        unset($this->routes[$name]);
+        $this->addWithPriority($name, $route, 0);
+    }
 
-        $this->routes[$name] = $route;
+    /**
+     * Adds a route.
+     *
+     * @param string $name  The route name
+     * @param Route  $route A Route instance
+     */
+    public function addWithPriority($name, Route $route, int $priority = 0)
```

Comment:
Instead of this, I would have used the deprecation mechanism with args and merged this with `addRoute` (only one entrypoint to add a route would then be used)

Reply:
I agree